api/agent/data_access.go

```diff
@@ -88,15 +88,15 @@ func (da *CachedDataAccess) GetAppByID(ctx context.Context, appID string) (*mode
 	return app.(*models.App), nil
 }
 
-func (da *CachedDataAccess) GetAppByName(ctx context.Context, app *models.App) (*models.App, error) {
-	key := appNameCacheKey(app.Name)
+func (da *CachedDataAccess) GetAppByName(ctx context.Context, appName string) (*models.App, error) {
+	key := appNameCacheKey(appName)
```

Comment:
get app by id should be the only one caching or we risk:

```
user A has app X with ID 1
user A deletes app X with ID 1
user B creates app X with ID 2
user B functions get filled with config from app X with ID 1
```

not sure if there are other places this comes up, think it's been mentioned in previous reviews though.